clients/object_store_client.go

```diff
@@ -93,6 +93,32 @@ func UploadToOSURLFields(osURL, filename string, data io.Reader, timeout time.Du
 	return nil
 }
 
+func CheckWritePermission(url *url.URL) error {
+	if url.Scheme == "memory" {
+		// delete is not supported for memory and it's only used for tests so just return here
+		return nil
+	}
+
+	tmpFileName := "tmp"
```

Comment:
Maybe add a random UUID or timestamp here to avoid concurrency issues with multiple nodes running this?

Reply:
I think we should be ok from that point of view since we're already writing files statically named e.g. 'index.m3u8' etc? For normal vod jobs this will be a location unique to the job and for transcode api the users have defined their output location.